src/bindings/cpp/include/key.hpp

```diff
@@ -238,6 +238,8 @@ class NameIterator
 public:
 	typedef std::string value_type;
 	typedef std::string reference;
+	typedef std::string pointer;
+	typedef int difference_type;
```

Comment:
in case someone asks: the types are required for SWIGs iterator class for python